docs/proposals/scheduling/20250217-node-resource-balance-rescheduling.md

```diff
@@ -0,0 +1,87 @@
+Over time, certain nodes in the cluster may experience imbalances in the allocation of different resources.
+This situation can cause resources with lower allocation rates in those nodes to become fragmented and underutilized, resulting in inefficient resource utilization and ultimately causing resource wastage.
+
+### Goals
+- Provides a configurable descheduling plugin to help rebalancing resource allocation at the node level.
+- Define the metrics used to represent the imbalance of different resources on a node, and define the threshold values for selecting nodes for eviction.
+- Define the eviction priority of Pods on the node.
+
+### Non-Goals/Future Work
+- Considering real utilization with Koordinator's NodeMetrics.
+- Trying to design a scheduling plug-ins to keep all nodes in resource balance as much as possible.
+- Providing best practices for working together with the Scheduler, e.g., using scheduling plugins like **NodeResourcesBalancedAllocation**.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+When the resource allocation rates of certain nodes in the cluster become imbalanced, resources with lower allocation rates on those nodes will become fragmented and underutilized. 
+For example, suppose node A has a CPU allocation rate of 90% and a memory allocation rate of 50%, while node B has a CPU allocation rate of 50% and a memory allocation rate of 90%. 
+If a pod requests 15% CPU and 15% memory of the node, the pod may fail to be scheduled, even though the total resources on node A and node B are sufficient.
+Such situations should be avoided.
+
+### Implementation Details
+
+#### Node Selection
+The node's fragmentation rate is described using the standard deviation of allocation rates across different resource types on the node:
+```node fragmentation rate = std(node CPU allocation rate, node memory allocation rate).```
+In scenarios involving only two resource types, the standard deviation simplifies to `|node CPU allocation rate - node memory allocation rate| / 2`. 
+This metric intuitively represents the balance degree of resource allocation rates of a node.
+It consistency with the scheduler's **NodeResourcesBalancedAllocation** strategy and provides scalability for incorporating additional resource types in the future.
+
+A node is considered to have excessive fragmentation when its fragmentation rate exceeds a threshold, which may adversely affect future pod scheduling on that node.
+To simplify user configuration, the plugin autonomously computes the mean (μ) and standard deviation (σ) of fragmentation rates across the cluster, dynamically setting the threshold at μ + σ.
+
+#### Pod Selection
+The plugin receives externally filter to evaluate and classify Pods on a node into two distinct categories: removable Pods and non-removable Pods. 
+Subsequently, a secondary filtering process is applied to the removable Pods, which eliminates those whose eviction would result in an increased fragmentation rate on the node.
```

Comment:
Besides, we need to judge if there is a suitable target node to place the evicted Pod. First, the target node must have enough available CPU and memory resources to accommodate the evicted Pod, this can refer to LowNodeLoad plugin inside koord-descheduler. Second, after placing the Pod on the target node, the node’s resource allocation should not become fragmented.

I feel this is a little bit complicated, or maybe I'm not very clear-minded. We can discuss further.

Reply:
And inside proposal, let's discuss all details clearly. Then we can implement only the most important features inside the first version of code.